bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -948,12 +1026,27 @@ public boolean setUrl(String url, String postData, String[] headers) {
 		if (stream != null) stream.Release();
 		if (hr != COM.S_OK) error(SWT.ERROR_NO_HANDLES, hr);
 		IUnknown request = new IUnknown(ppRequest[0]);
-		hr = webView_2.NavigateWithWebResourceRequest(request);
-		request.Release();
+		lastWebView_2Task = acceptAsyncAndRunSync(lastWebView_2Task, wv -> {
+			wv.NavigateWithWebResourceRequest(request);
+			request.Release();
+		});
 	} else {
-		hr = webView.Navigate(pszUrl);
+		lastWebViewTask = acceptAsyncAndRunSync(lastWebViewTask, wv -> wv.Navigate(pszUrl));
 	}
-	return hr == COM.S_OK;
+	return true;
+}
+
+private <T extends ICoreWebView2> CompletableFuture<T> acceptAsyncAndRunSync(CompletableFuture<T> webViewTask, Consumer<T> action) {
```

Comment:
```suggestion
private <T extends ICoreWebView2> CompletableFuture<T> consumeAndResizeBrowser(CompletableFuture<T> webViewTask, Consumer<T> action) {
```

Reply:
The Resize in the method is only to trigger repainting since the shell doesnt seem to paint the browser if it is not initialized. Hence the name resize is not really relevant for this method.

Reply:
Renamed to `scheduleWebViewTask`